stdlib/list.ml

```diff
@@ -56,6 +56,14 @@ let rec rev_append l1 l2 =
 
 let rev l = rev_append l []
 
+let init n f =
+  if n = 0 then [] else
+  if n < 0 then invalid_arg "List.init" else
+  let rec init_aux acc i =
+    if i >= n then acc else
+    init_aux ((f i)::acc) (i+1)
+  in rev (init_aux [] 0)
```

Comment:
I wondered whether really we should be adding scan_left and scan_right at this point, but that's a bit too SML-y for the OCaml standard library! It seems better to me to maintain the duality with `Array.init` - so document the fact.

That said, `Array.init` doesn't note it either...